apps/bfd-model/bfd-model-rif/src/main/resources/db/scripts/BFD-1010-schema-changes/migrate_data.sh

```diff
@@ -1,67 +1,113 @@
-#!/bin/bash
+#!/usr/bin/env bash
+set -eo pipefail
+export MAX_JOBS="${MAX_JOBS:-1}" # defaults to one, probably max around 4
+export PGHOST="${PGHOST:-}"
+export PGUSER="${PGUSER:-}"
+export PGPASSWORD="${PGPASSWORD:-}"
+export PGPORT="${PGPORT:-5432}"
+export PGDATABASE="${PGDATABASE:-}"
 
-set -o pipefail
-set -e
+# we will process these one by one in order (I do not know which tables are parents, just guessing)
+parent_tables=(beneficiaries beneficiaries_history)
 
-# following must be passed in as either environment variables or as cmd-line args (default)
-PGHOST="${DB_HOST:-$1}"
-PGUSER="${DB_USER:-$2}"
-PGPASSWORD="${DB_PSWD:-$3}"
+# these we will run in background jobs, up to $MAX_JOBS at a time
+child_tables=(
```

Comment:
The only trouble here is that the db is (sort of) made up of 3 relationship tiers:

- _beneficiaries_ (bene_id is a constraint for pretty much all tables)
- then top-level _claims_ tables come next (i.e., bene_id will be contained in each claims table)
- then claim line items will need their _parent claims_ rcd to exist

All the other ancillary bene tables (beneficiary_monthly, etc.) can run pretty much anytime after the beneficiaries table is populated. I'll look into creating that within the script but might holler for help as needed. Thanks for doing this...it's a good model to have for db stuff.